flink-connector-mongodb/src/test/java/org/apache/flink/connector/mongodb/table/converter/MongoConvertersTest.java

```diff
@@ -146,28 +146,31 @@ public void testConvertBsonToRowData() {
                 GenericRowData.of(
                         StringData.fromString(oid.toHexString()),
                         StringData.fromString("string"),
-                        StringData.fromString(uuid.toString()),
+                        StringData.fromString(
+                                "{\"_value\": {\"$binary\": {\"base64\": \"gR+qXamERr+L0IyxvO9daQ==\", \"subType\": \"04\"}}}"),
                         2,
                         3L,
                         4.1d,
                         DecimalData.fromBigDecimal(new BigDecimal("5.1"), 10, 2),
                         false,
                         TimestampData.fromEpochMillis(now.getEpochSecond() * 1000),
+                        TimestampData.fromEpochMillis(now.toEpochMilli()),
                         StringData.fromString(
-                                OffsetDateTime.ofInstant(
-                                                Instant.ofEpochMilli(now.toEpochMilli()),
-                                                ZoneOffset.UTC)
-                                        .format(ISO_OFFSET_DATE_TIME)),
-                        StringData.fromString("/^9$/i"),
-                        StringData.fromString("function() { return 10; }"),
-                        StringData.fromString("function() { return 11; }"),
-                        StringData.fromString("12"),
-                        StringData.fromString(oid.toHexString()),
+                                "{\"_value\": {\"$regularExpression\": {\"pattern\": \"^9$\", \"options\": \"i\"}}}"),
```

Comment:
shouldn't the sting be just `{\"$regularExpression\": {\"pattern\": \"^9$\", \"options\": \"i\"}}` without the outer `_value`? or is this important for the round trip?

Reply:
Thanks @twalthr.
The `JsonWriter` cannot directly write a `BsonValue` in to a string. It will throw an exception when writing directly to a `BsonValue`, so we used `_value` to wrap the bson value into a bson document. However, we can also extend a `JsonWriter` so that it does not check when writing bson value directly.  Do you think we need to customize a `JsonWriter`?

```java
package org.apache.flink;

import org.bson.BsonRegularExpression;
import org.bson.json.JsonWriter;

import java.io.IOException;
import java.io.StringWriter;

public class JsonWriterTest {

    public static void main(String[] args) throws IOException {
        try (StringWriter stringWriter = new StringWriter();
             JsonWriter jsonWriter = new JsonWriter(stringWriter)) {
            BsonRegularExpression regularExpression = new BsonRegularExpression("regex", "i");
            jsonWriter.writeRegularExpression(regularExpression);
        }
    }
}
```

```shell
Exception in thread "main" org.bson.BsonInvalidOperationException: A RegularExpression value cannot be written to the root level of a BSON document.
	at org.bson.AbstractBsonWriter.throwInvalidState(AbstractBsonWriter.java:740)
	at org.bson.AbstractBsonWriter.checkPreconditions(AbstractBsonWriter.java:701)
	at org.bson.AbstractBsonWriter.writeRegularExpression(AbstractBsonWriter.java:590)
	at org.apache.flink.JsonWriterTest.main(JsonWriterTest.java:15)
```

Reply:
The question whether we need to customize a JsonWriter depends on the round-trip story. Is it possible to read a RegExp as string and write it out as a string again that MongoDB could correctly classify as RegExp again?

Reply:
The current bson api only supports the conversion of bson document. For a single bson value, we need to customize `JsonWriter` and `JsonReader`.
So in the previous implementation, we used a _value to wrap a single bson value as a bson document, so that we can parse them easily.

```java
import org.bson.BsonDocument;
import org.bson.BsonRegularExpression;
import org.bson.json.JsonMode;
import org.bson.json.JsonWriterSettings;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class JsonConversionTest {

    @Test
    public void bsonToJsonTest() {
        BsonRegularExpression original = new BsonRegularExpression("regex", "i");
        BsonDocument wrapped = new BsonDocument("_value", original);

        String json = wrapped.toJson(JsonWriterSettings.builder().outputMode(JsonMode.EXTENDED).build());
        // {"_value": {"$regularExpression": {"pattern": "regex", "options": "i"}}}
        System.out.println(json);

        BsonDocument parsed = BsonDocument.parse(json);
        BsonRegularExpression parsedRegularExpression = parsed.getRegularExpression("_value");

        assertEquals(parsedRegularExpression, original);
    }
}
```

Reply:
Hi @twalthr.
In a recent commit, read a RegExp as string and write it out as a RegExp to MongoDB again is supported.
Please help to review it when you have time.